packages/codegen.go/src/models.ts

```diff
@@ -372,7 +372,17 @@ function generateJSONMarshallerBody(modelType: go.ModelType | go.PolymorphicType
       } else {
         modelDef.SerDe.needsJSONPopulate = true;
       }
-      marshaller += `\t${populate}(objectMap, "${field.serializedName}", ${receiver}.${field.name})\n`;
+      if (go.isPrimitiveType(field.type) && (field.type.typeName.startsWith('uint') || field.type.typeName.startsWith('int')) && field.type.encodeAsString) {
```

Comment:
Looks like array of these is missing in both cases. If there is no test coverage for that it's ok to omit for now but will need to be addressed at some point. What services are asking for this?

Reply:
Batch is asking for this. It seems I also miss the map, let me check the logic.